base/utf16.jl

```diff
@@ -66,15 +76,190 @@ function encode16(s::AbstractString)
     UTF16String(buf)
 end
 
+function ascii_to_utf16(s::Array{UInt8,1})
+    len::Int = sizeof(s)
+    buf = Array{UInt16,1}(len+1)
+    for i = 1:len
```

Comment:
Typing `copy!` in the search field at docs.julialang.org worked for me.  You can also type `?copy!` at the REPL.
Regarding the second question, this is also pretty straightforward to check yourself at the REPL.